### ISSUE TITLE
CLI Design

### ISSUE THREAD

**Issue by Danappelxx** [QuarkX/Quark#6]
Before we can start developing the CLI, we have to agree on a design. I was thinking of a `command:subcommand` system syntax where subcommands are denoted with semicolons.

Examples:
```shell
quark deploy:heroku
```
```shell
quark new:resource:name="Todo"
```
```shell
quark xcode
```
```shell
quark spm:add:name="Zewo/PostgreSQL",version:"0.7"
```

I'm undecided for how to handle arguments. `command:subcommand --arg=value --arg2=value2` or with `command:subcommand:arg=value,arg2=value2`. Both feel a little verbose to me, so I'm not sure if I like either.

RFC!

**Comment by paulofaria** [comment]
@unnamedd told me he was thinking about this syntax as well. personally I prefer just:

```
quark new app AppName
quark new resource ResourceName
quark deploy heroku
quark new package Zewo/PostgreSQL 0.7
```

keep it simple

**Comment by Danappelxx** [comment]
Maybe we should start with that for simplicity, but I'd like to have a "modern" feel. I was drawing inspiration from [Mac-CLI](https://github.com/guarinogabriel/Mac-CLI), which I felt was a pleasure to use.

**Comment by Danappelxx** [comment]
What about

```shell
quark new:app AppName
quark new:resource ResourceName
quark deploy heroku
quark add:package Zewo/PostgreSQL 0.7
```

where we still have the `command:subcommand`, but parameters are passed without labels (for terseness).

**Comment by paulofaria** [comment]
that's closer to what @unnamedd proposed I think. Still prefer whitespace as separator haha. what's the advantage you see in using `:`? Heroku also uses `:`.

**Comment by paulofaria** [comment]
I'm not sure it's cleaner. Whitespace looks cleaner to me. I think more important than the style is the actual commands. let's decide the core commands before. and then we can see which style fits best with all commands.

```
quark new AppName
quark new resource ResourceName
quark new service ServiceName
quark xcode
quark build
quark test
quark run
quark deploy
```

The deploy wouldn't need a parameter cause it would pick configurations from a file.

**Comment by unnamedd** [comment]
I am not opposing me against use whitespace as separator and maybe in this moment, when we do not have many commands and subcommands, use whitespace should be the simplest way to design the CLI commands, but in the near future, we will have many subcommands and it could be better use some convention which show to user as soon as he seems that we have more subcommands under that command.

`$ quark new AppName` is good, but
`$ quark new:resource resourceName` is better than `

**Comment by paulofaria** [comment]
Yeah! That's why I said we should have a clear vision of all the commands we're building. We should create a living document with the proposed commands and their descriptions. Then we discuss then and when we have all the commands mapped we decide the style.

**Comment by paulofaria** [comment]
I think the best approach is to use the issue description. Everything who's a member of the organization can edit it. So I'll just add a section below the description with the proposal and we can discuss on the comments here and if anyone wants to add something just edit the issue description.
